jdbc/src/main/java/io/kestra/jdbc/repository/AbstractJdbcRepository.java

```diff
@@ -346,6 +346,24 @@ protected Condition getConditionOnField(
         };
     }
 
+    private static Object primitiveOrToString(Object o) {
+        if (o == null) return null;
+
+        if (o instanceof Boolean
+            || o instanceof Byte
+            || o instanceof Short
+            || o instanceof Integer
+            || o instanceof Long
+            || o instanceof Float
+            || o instanceof Double
+            || o instanceof Character
+            || o instanceof String) {
+            return o;
+        }
+
+        return Objects.toString(o);
```

Comment:
No need to something more complex as you already check for nullability.

```suggestion
        return o.toString();
```